src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,52 +1,131 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
+ * <p>Реалізувати свій ArrayList який імплементує інтерфейс List.</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private Object[] data;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            data = new Object[initialCapacity];
+        } else {
+            throw new IllegalArgumentException("Wrong initial capacity: " + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        data = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity(int amount) {
+        if (data.length <= amount) {
+            data = Arrays.copyOf(data, data.length + amount + DEFAULT_CAPACITY);
+        }
+    }
+
+    private void arrayIndexOutOfBoundsException(int index) {
+        throw new ArrayIndexOutOfBoundsException("The index is out of range: "
+                + index + ", the size is: " + this.size());
+    }
 
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        size++;
+        data[size - 1] = value;
```

Comment:
```suggestion
        data[size++] = value;
```and then remove         size++;